core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -58,6 +58,11 @@ private case object DoPutSucceeded extends DoPutResult
 private case object DoPutBytesFailed extends DoPutResult
 private case class DoPutIteratorFailed(iter: Iterator[Any]) extends DoPutResult
 
+private[spark] object BlockManagerConfiguration {
+  // Max number of failures before this block manager refreshes the block locations from the driver
+  def maxFailuresBeforeLocationRefresh: Int = 5
+}
```

Comment:
I don't think we really need a separate config object for this, you could put it in the existing `object BlockManager`